ITE/2/README.adoc

```diff
@@ -0,0 +1,473 @@
+manner:
+
+[arabic]
+. The root of trust for all packages, as well as TUF and in-toto
+metadata.
+. The software supply chain defined using in-toto.
+. The public keys used to verify this supply chain.
+
+TUF also guarantees that MitM attackers cannot tamper with the
+consistency, authenticity, and integrity of these files, nor rollback or
+indefinitely replay metadata. The key point is that this offline
+bootstrapping of trust with TUF is what gives in-toto meaningful
+security guarantees.
+
+== Specification
+
+[[basic-security-model]]
+=== Basic security model
+
+Neither in-toto or TUF mandate a particular security model for implementation.
+That is, it is possible for a deployment to use only online
+keys and a single online-signed layout to secure their supply chain.
+However, such a deployment will have marginal security benefits. In
+contrast, the security model proposed below aims to provide strong
+security guarantees in the event of infrastructure compromise,
+misbehaving functionaries, and even project owners, while maintaining
+reasonable operational overhead in most cases.
+
+This section discusses how the reader SHOULD combine TUF and in-toto in
+order to build-compromise resilience in general, with no particular CI/CD
+pipeline in mind. The reader is assumed to be familiar with the concepts and
+terminology of public-key cryptography as well as to have thoroughly read and
+understood the
+https://github.com/theupdateframework/specification[TUF] and
+https://github.com/in-toto/docs[in-toto] specifications. Finally, the
+reader is also assumed to have access to implementations of TUF (e.g.,
+for https://github.com/theupdateframework/tuf[Python],
+https://github.com/theupdateframework/notary[Docker],
+https://github.com/flynn/go-tuf[Go], and
+https://github.com/heartsucker/rust-tuf[Rust]) and in-toto (e.g., for
+https://github.com/in-toto/in-toto[Python],
+https://github.com/in-toto/in-toto-java[Java], and
+https://github.com/in-toto/in-toto-golang[Go]) that are capable of
+meeting the requirements outlined in this section.
+
+This subsection describes the requirements that MUST be met in order for
+an integration of TUF and in-toto to be said to be compromise-resilient.
+
+The in-toto root layout MUST consist of at least two steps:
+
+[arabic]
+. The "`developer`" step is performed by a developer who signs source
+code. Typically, it receives no materials, and outputs source code as
+products. Its link metadata MUST be signed by developers using (m, n)
+offline keys, where m >= 1, and n is the number of currently trusted
+developers. It is OPTIONAL but RECOMMENDED to protect these keys using
+hardware such as https://github.com/DataDog/yubikey[YubiKeys].
+. The "`packager`" step is performed by the CI/CD pipeline which
+transforms this source code into packages (e.g., tarballs, binaries).
+Typically, it receives source code from the "`developer`" step, and
+outputs packages as products. It SHOULD use (1, 1) online keys.
```

Comment:
Talking about the rationale here (why SHOULD, when to use other values, etc.) may be helpful.